src/_parser.ts

```diff
@@ -26,18 +27,22 @@ function toVal(out, key, val, opts) {
   const old = out[key];
   const nxt = ~opts.string.indexOf(key)
     ? val == undefined || val === true
-      ? ""
+      ? undefined
       : String(val)
-    : typeof val === "boolean"
-      ? val
-      : ~opts.boolean.indexOf(key)
-        ? val === "false"
-          ? false
-          : val === "true" ||
-            (out._.push(((x = +val), x * 0 === 0) ? x : val), !!val)
-        : ((x = +val), x * 0 === 0)
-          ? x
-          : val;
+    : ~opts.number.indexOf(key)
```

Comment:
`~-1` == `0` while is valid, is really not readable

Reply:
Done.

I replaced `indexOf` with `includes` and also updated other instances of `~`.